src/pulselistener/pulselistener/listener.py

```diff
@@ -60,48 +62,52 @@ def __init__(self, configuration, bus):
     def is_coverage_task(self, task):
         return any(task['task']['metadata']['name'].startswith(s) for s in ['build-linux64-ccov', 'build-win64-ccov'])
 
-    def get_build_task_in_group(self, group_id):
+    async def get_build_task_in_group(self, group_id):
         if group_id in self.triggered_groups:
             logger.info('Received duplicated groupResolved notification', group=group_id)
             return None
 
         def maybe_trigger(tasks):
+            logger.info('Checking code coverage tasks', group_id=group_id, nb=len(tasks))
             for task in tasks:
                 if self.is_coverage_task(task):
                     self.triggered_groups.add(group_id)
                     return task
 
             return None
 
-        def retrieve_coverage_task(limit=200):
-            reply = self.queue.listTaskGroup(
-                group_id,
-                limit=limit,
-            )
-            task = maybe_trigger(reply['tasks'])
+        def load_tasks(limit=200, continuationToken=None):
+            query = {
+                'limit': limit,
+            }
+            if continuationToken is not None:
+                query['continuationToken'] = continuationToken
+            reply = retry(lambda: self.queue.listTaskGroup(group_id, query=query))
+            return maybe_trigger(reply['tasks']), reply.get('continuationToken')
 
-            while task is None and reply.get('continuationToken') is not None:
-                reply = self.queue.listTaskGroup(
-                    group_id,
-                    limit=limit,
-                    continuationToken=reply['continuationToken'],
-                )
-                task = maybe_trigger(reply['tasks'])
+        async def retrieve_coverage_task():
+            task, token = load_tasks()
+
+            while task is None and token is not None:
+                task, token = load_tasks(continuationToken=token)
+
+                # Let other tasks run on long batches
+                await asyncio.sleep(2)
```

Comment:
You can use sleep(0) to allow other tasks to run.

(It would be even better to use the async version of the Taskcluster lib, not sure how easy it is though)

Reply:
I tried to use the async lib... and failed (not given much time). It's there as `aio` in the tc lib; but no example